### PR TITLE
Pool rodent - Private Collections

### PR DESCRIPTION
**Pull request by sarahzinger**
## Links
* Remix link: https://pool-rodent.glitch.me/
* Issue-tracker link: this pr covers several tickets in the [Private Collections epic](https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/board?assignees=sarahzinger&epics=192398789_427&repos=117580951,192398789) including: 
    - https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/431
    - https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/433
    - https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/432
* Specs/doc links: https://docs.google.com/document/d/1E-gJwFbmOUxwEAYLIGLNWFJwZsOXLSvBPhJuCN80UHA/edit#bookmark=id.xrq7m23ovvia

## GIF/Screenshots:
Screen Shot 2019-09-17 at 11 26 47 AM
Screen Shot 2019-09-17 at 11 27 35 AM
Screen Shot 2019-09-17 at 11 27 47 AM
Screen Shot 2019-09-17 at 11 14 38 AM

## Changes:
Adds Private Collections to the frontend and updates:
- the collections list on a user page, to have a view only privacy badge on private collections
- the collection page itself to have a privacy toggle
- the add project to collections popover to make private collection collections yellow

## How To Test:
* unfortunately the backend for this has yet to be deployed: https://github.com/FogCreek/Glitch/pull/1924 so testing is tricky unless you know how to run the backend locally (which is very possible and I can attempt to remember how to set it up if you want i can try to show ya! but it's definitely a little bit of a pita right now), we may wish to wait until that's shipped before we ship this although it's not necessary because...
* I have written this in such a way that it should all be properly darkshipped behind the `My Stuff` feature flag so as long as you don't have that checked everything should look exactly as it used to be which is good to test.

## Feedback I'm looking for:
- anything I'm missing? any style bugs? One thing that occurs to me is I'm not sure it's clear to users who use a screenreader if private collections (or projects!) in an add project to collection popover are private, as I think the only thing we do is make the background yellow. Perhaps I should abstract this into `<Result Item />` as a private prop that changes the background and maybe adds a visually hidden word of "Private" the whatever the screenreader announces? I was thinking though that that might be something to do after we've incorporated the ResultItems from `shared-components` in a separate pr. Let me know your thoughts and I can make a card for that work if it belongs elsewhere, or if we want to handle it here I can update this pr directly :)

### CODE REVIEW

File: .glitch-assets [hunk 1]
```diff
@@ -2125,3 +2125,4 @@
 {"name":"2fa.svg","date":"2019-09-12T17:35:05.782Z","url":"https://cdn.glitch.com/02863ac1-a499-4a41-ac9c-41792950000f%2F2fa.svg","type":"image/svg+xml","size":12080,"imageWidth":108,"imageHeight":122,"thumbnail":"https://cdn.glitch.com/02863ac1-a499-4a41-ac9c-41792950000f%2F2fa.svg","thumbnailWidth":108,"thumbnailHeight":122,"uuid":"ylYbelBVKvsWFGse"}
 {"uuid":"JUZyjdf9HexFWK0m","deleted":true}
 {"name":"magic-link.svg","date":"2019-09-16T18:24:25.397Z","url":"https://cdn.glitch.com/0aa2fffe-82eb-4b72-a5e9-444d4b7ce805%2Fmagic-link.svg","type":"image/svg+xml","size":10921,"imageWidth":84,"imageHeight":108,"thumbnail":"https://cdn.glitch.com/0aa2fffe-82eb-4b72-a5e9-444d4b7ce805%2Fmagic-link.svg","thumbnailWidth":84,"thumbnailHeight":108,"uuid":"V75f5YfKyXXPcO7W"}
+{"name":"brainfriends.svg","date":"2019-09-18T13:50:32.916Z","url":"https://cdn.glitch.com/0aa2fffe-82eb-4b72-a5e9-444d4b7ce805%2Fbrainfriends.svg","type":"image/svg+xml","size":49171,"imageWidth":659,"imageHeight":419,"thumbnail":"https://cdn.glitch.com/0aa2fffe-82eb-4b72-a5e9-444d4b7ce805%2Fthumbnails%2Fbrainfriends.svg","thumbnailWidth":330,"thumbnailHeight":210,"uuid":"xJ9NNFX0df5PP8Wq"}
```
Comment (sarahzinger) [inline .glitch-assets]: i think this is from me pulling in master

**Comment by sarahzinger** [comment]
Ok it's not perfect, but I think it might be "good enough". In particular I ran into some css struggles in making this and probably did not do it in the most eloquent method, but I think it's probably ok for now.

My Stuff Collections list, now private by default, some small fixes here:
Screen Shot 2019-09-20 at 9 10 11 AM

Private Collections look different in the collections list
Screen Shot 2019-09-20 at 9 10 41 AM

Private Collection pages now with a left aligned tooltip:
Screen Shot 2019-09-20 at 9 11 05 AM

Public Collection pages have a privacy toggle have a background color now so they don't fade into the background too much
Screen Shot 2019-09-20 at 9 11 22 AM

Project Page Privacy Toggles still work and My Stuff button's checkmark is properly hidden again
Screen Shot 2019-09-20 at 9 11 30 AM
Screen Shot 2019-09-20 at 9 11 41 AM

**Comment by ttseng** [comment]
the privacy button on profile pages seems a bit sensitive - it seems like the cursor on the label is interfering with cursor events on its parent span:

pointer-cursor

I see that if I turn off pointer events on the label (`.hidden-checkbox_label {pointer-events: none;}`), it's a lot smoother:

label-pointer-events-none

**Comment by ttseng** [comment]
my bad - I made a mistake in my spec.  It looks like the privacy badge in collection name buttons don't match the project ones:

image

because of the border around the privacy badge for collections, it looks like there's a button in a button.  we should use the same styling we have for projects.

File: src/components/private-badge/styles.styl [hunk 2]
```diff
@@ -8,17 +8,21 @@
...
   border-radius: 3px
   vertical-align: middle
-  >*
-    left: 5px
-    top: 2px
+  svg
+    margin: 2px 4px
   &.private
     background-color: private
+    border: 1px solid private-foreground
```
Comment (ttseng) [inline src/components/private-badge/styles.styl]: to match project privacy badge, we can change this to

```
&.private
color: private
```

(remove background-color and border attributes)

**Comment by ttseng** [comment]
In adding projects to a collection, I'm seeing the privacy background color for result items flash before returning back to white. 

(In the example below, My Stuff should stay yellow until you hover over it)

my-stuff-privacy-flash

**Comment by sarahzinger** [comment]
@ttseng thanks for the review!! mind taking another look for me? I believe I removed the flicker you're seeing on the add project to collection popover, but the reason the background is not white is because algolia search results on collections are not returning a `private` property on collections (I'll make sure to update that on the backend!), the reason it was sometimes before is because we show a fake My Stuff collection if you don't have one yet, and before I would add it before the collections finished loading, that shouldn't be the case now. 

I couldn't add `pointer-events: none` to the section you were suggesting because that one is a button/not a badge, but I made some other style changes so it's hopefully a smoother experience!

**Privacy Badge on Project Pages** 
(this one is a change from our current look, let me know if it seems weird! You would only get here if you go to a private team project that you are not a member of)
Screen Shot 2019-09-23 at 4 41 16 PM

**Privacy Toggle on Project Pages**
Screen Shot 2019-09-23 at 4 41 51 PM
Screen Shot 2019-09-23 at 4 41 57 PM

**Privacy Badge on Collections List**
Screen Shot 2019-09-23 at 4 42 05 PM

**Privacy Button on Collection Page**
Screen Shot 2019-09-23 at 4 42 11 PM
Screen Shot 2019-09-23 at 4 42 16 PM

**Comment by sarahzinger** [comment]
ok @ttseng the backend is deployed and I pulled in changes from master, I believe this should be ready to look at again!

**Comment by ttseng** [comment]
Nice - seeing the privacy styling on collection result items now!  
image 

Just had a comment about the sensitivity of the privacy button on collection pages but otherwise I think this is good to go!

File: src/components/private-badge/styles.styl [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
+.privateButton
   border: 1px solid secondary
   cursor: pointer
+  svg 
+    margin: 1px 0
+    cursor: pointer
   >*
-    left: 4px
\ No newline at end of file
+    left: 4px
```
[the remark this answers is not in the document]
Comment (sarahzinger) [reply src/components/private-badge/styles.styl]: rad fixed!

**Pull request merged**